tests/nvapi_drs.cpp

```diff
@@ -50,6 +104,22 @@ TEST_CASE("DRS methods succeed", "[.drs]") {
         REQUIRE(NvAPI_DRS_CreateProfile(handle, &profileInfo, &profile) == NVAPI_NOT_SUPPORTED);
     }
 
+    SECTION("GetSetting with unknown struct version returns incompatible-struct-version") {
+        NvDRSSessionHandle handle{};
+        NvDRSProfileHandle profile{};
+        NVDRS_SETTING setting;
+        setting.version = NVDRS_SETTING_VER + 1;
+        REQUIRE(NvAPI_DRS_GetSetting(handle, profile, FXAA_ALLOW_ID, &setting) == NVAPI_INCOMPATIBLE_STRUCT_VERSION);
+    }
+
+    SECTION("GetSetting with current struct version returns not incompatible-struct-version") {
+        NvDRSSessionHandle handle{};
+        NvDRSProfileHandle profile{};
+        NVDRS_SETTING setting;
+        setting.version = NVDRS_SETTING_VER;
+        REQUIRE(NvAPI_DRS_GetSetting(handle, profile, FXAA_ALLOW_ID, &setting) != NVAPI_INCOMPATIBLE_STRUCT_VERSION);
+    }
+
     SECTION("GetSetting returns setting-not-found") {
```

Comment:
Would be cool to have a test for `GetSetting` that uses `DXVK_NVAPI_DRS_SETTINGS`.  That said, I guess reading the env var needs to move to `NvAPI_DRS_GetSetting` (or e.g. `NvAPI_DRS_CreateSession`) for that. Since this is not really a hot path, I would be fine with the small penalty.

PS: Resetting env vars between tests happens in https://github.com/jp7677/dxvk-nvapi/blob/master/tests/section_listener.h#L11

Reply:
And here I liked being able to make the map `static const`… :sweat_smile:

Reply:
If it’s ok move the declaration into NvAPI_DRS_GetSetting, static const would still be ok, as long as the env var is set before NvAPI_DRS_GetSetting is being tested. No need to have more than one variation in the tests (but a short comment would be nice noting that there is only one shot)